src/drivers/lights/rgbled_lp5562/rgbled_lp5562.cpp

```diff
@@ -0,0 +1,320 @@
+
+	LedController _led_controller;
+	uint8_t _current = 175; // matching default current of 17.5mA
+};
+
+RGBLED_LP5562::RGBLED_LP5562(const I2CSPIDriverConfig &config) :
+	I2C(config),
+	I2CSPIDriver(config)
+{
+	_current = config.custom1;
+}
+
+int
+RGBLED_LP5562::init()
+{
+	int ret = I2C::init();
+
+	if (ret != OK) {
+		return ret;
+	}
+
+	uint8_t command[1] = {ENABLE_CHIP_EN};
+	ret = write(ENABLE_ADDR, command, sizeof(command));
+
+	if (ret != OK) {
+		return ret;
+	}
+
+	// We have to wait 500us after enable.
+	px4_usleep(500);
+
+	command[0] = CONFIG_ENABLE_INTERNAL_CLOCK;
+	ret = write(CONFIG_ADDR, command, sizeof(command));
+
+	if (ret != OK) {
+		return ret;
+	}
+
+	command[0] = LED_MAP_ALL_PWM;
+	ret = write(LED_MAP_ADDR, command, sizeof(command));
+
+	if (ret != OK) {
+		return ret;
+	}
+
+	// Write all 3 colors at once.
+	uint8_t currents[3] = {_current, _current, _current};
+	ret = write(B_CURRENT_ADDR, currents, sizeof(currents));
+
+	if (ret != OK) {
+		return ret;
+	}
+
+	ScheduleNow();
+
+	return OK;
+}
+
+int
+RGBLED_LP5562::probe()
+{
+	uint8_t result[1] = {0};
+	int ret = read(W_CURRENT_ADDR, result, sizeof(result));
+
+	if (ret != OK) {
+		return ret;
+	}
+
+	_retries = 1;
+
+	return (result[0] == W_CURRENT_DEFAULT) ? OK : ERROR;
+}
+
+int
+RGBLED_LP5562::read(uint8_t address, uint8_t *data, unsigned count)
+{
+	uint8_t cmd = address;
+	return transfer(&cmd, 1, (uint8_t *)data, count);
+}
+
+int
+RGBLED_LP5562::write(uint8_t address, uint8_t *data, unsigned count)
+{
+	uint8_t buf[4];
+
+	if (sizeof(buf) < (count + 1)) {
+		return -EIO;
+	}
+
+	buf[0] = address;
+	memcpy(&buf[1], data, count);
+
+	return transfer(&buf[0], count + 1, nullptr, 0);
+}
+
+void
+RGBLED_LP5562::RunImpl()
+{
+	if (should_exit()) {
+		send_led_rgb(0, 0, 0);
+		return;
+	}
+
+	LedControlData led_control_data;
+
+	if (_led_controller.update(led_control_data) == 1) {
+
+		const uint8_t on = led_control_data.leds[0].brightness;
+
+		switch (led_control_data.leds[0].color) {
+		case led_control_s::COLOR_RED:
+			send_led_rgb(on, 0, 0);
+			break;
+
+		case led_control_s::COLOR_GREEN:
+			send_led_rgb(0, on, 0);
+			break;
+
+		case led_control_s::COLOR_BLUE:
+			send_led_rgb(0, 0, on);
+			break;
+
+		case led_control_s::COLOR_AMBER: // same as yellow
+		case led_control_s::COLOR_YELLOW:
+			send_led_rgb(on, on, 0);
+			break;
+
+		case led_control_s::COLOR_PURPLE:
+			send_led_rgb(on, 0, on);
+			break;
+
+		case led_control_s::COLOR_CYAN:
+			send_led_rgb(0, on, on);
+			break;
+
+		case led_control_s::COLOR_WHITE:
+			send_led_rgb(on, on, on);
+			break;
+
+		case led_control_s::COLOR_OFF:
+		default:
+			send_led_rgb(0, 0, 0);
+			break;
+		}
+
+	}
+
+	ScheduleDelayed(_led_controller.maximum_update_interval());
+}
+
+int
+RGBLED_LP5562::send_led_rgb(uint8_t r, uint8_t g, uint8_t b)
+{
+	uint8_t leds[3] = {b, g, r};
+	return write(B_PWM_ADDR, leds, sizeof(leds));
+}
+
+void
+RGBLED_LP5562::print_usage()
+{
```

Comment:
It would be good to be specific about products it is used it. This is my inference/the minimum a user might want to know.

```suggestion
{
	PRINT_MODULE_DESCRIPTION(
		R"DESCR_STR(
### Description

Driver for [LP5562](https://www.ti.com/product/LP5562) LED driver connected via I2C.
This used in some GPS modules for [PX4 status notification](../getting_started/led_meanings.md)

The driver is included by default in firmware (KConfig key DRIVERS_LIGHTS_RGBLED_LP5562) and is always enabled.

)DESCR_STR");
```

Reply:
Thanks that's good!